src/mixins/filing-template-mixin.ts

```diff
@@ -560,21 +569,28 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
 
     // Restore business information
     this.setBusinessInformation({
-      ...filing.business,
-      ...filing.changeOfRegistration.business
+      ...entitySnapshot.businessInfo
     })
 
+    // Restore Naics
+    if (filing.changeOfRegistration.business.naics) {
+      this.setNaics(filing.changeOfRegistration.business.naics)
+    }
```

Comment:
I'm not sure if you are referring to the condition or statement. 
In the condition there is a chance yes, since we are dealing with the draft filing here, if no changes were made, `naics` would be undefined. 
Inside the statement, it will be defined.

Reply:
Oh i see.
In this case we are dealing with a draft filing, so if `naics` is present in the draft it will restore it, otherwise we just continue on. 
We are able to do so because we already have the snapshot in store (always), so no fallback is required here.

Reply:
I suppose it would but we don't `setNaics` if `naics` is absent.

Reply:
Doh! You're right. 👍 

In that case, it keeps its default store/state value then (which I don't see !?).

Reply:
You betcha!
The snapshot and business info is restored at the start of the method